tla-assignments/src/main/scala/at/forsyte/apalache/tla/assignments/passes/PrimingPassImpl.scala

```diff
@@ -1,20 +1,24 @@
 package at.forsyte.apalache.tla.assignments.passes
 
+import at.forsyte.apalache.infra.passes.DerivedPredicates
 import at.forsyte.apalache.infra.passes.Pass.PassResult
-import at.forsyte.apalache.infra.passes.PassOptions
-import at.forsyte.apalache.tla.lir._
 import at.forsyte.apalache.io.lir.TlaWriterFactory
+import at.forsyte.apalache.tla.lir.UntypedPredefs._
+import at.forsyte.apalache.tla.lir._
 import at.forsyte.apalache.tla.lir.storage.BodyMapFactory
-import at.forsyte.apalache.tla.lir.transformations.{decorateWithPrime, TransformationTracker}
+import at.forsyte.apalache.tla.lir.transformations.TransformationTracker
+import at.forsyte.apalache.tla.lir.transformations.decorateWithPrime
 import at.forsyte.apalache.tla.lir.transformations.standard._
-import at.forsyte.apalache.tla.lir.UntypedPredefs._
 import com.google.inject.Inject
 import com.typesafe.scalalogging.LazyLogging
 
 /**
  * PrimingPass adds primes to the variables in state initializers and constant initializers.
  */
-class PrimingPassImpl @Inject() (options: PassOptions, tracker: TransformationTracker, writerFactory: TlaWriterFactory)
+class PrimingPassImpl @Inject() (
+    derivedPreds: DerivedPredicates,
```

Comment:
I'm not sure I follow your question, but if you're asking why this is called `derivedPreds` here, when we aren't actually deriving them or anything, its just so that we have consistent naming in all the passes that access this data structure.

Reply:
I meant that the only kind of predicates at this point will be `DerivedPredicates` at this point, right?
If so, it doesn't add much – rather, might be confusing – to distinguish it by name. But that's maybe just my preference.

Reply:
Do you think this would be better as

```suggestion
    specPredicats: DerivedPredicates,
```

?

Reply:
I will lave this for potential followup improvement. I don't think it will cause confusion as is.

Reply:
Totally, we can keep it as-is!